### ISSUE TITLE
Feature request: Support for private / non-default repositories

### ISSUE THREAD

**Issue by aripalo** [eladb/projen#124]
## Problem

Currently it seems that Projen only supports the public repositories (npmjs.com, pypi.org, maven central). 

This is controlled by `jsii-project.ts`:
- for [NPM in lines 299-300](https://github.com/eladb/projen/blob/ef47eebc3115a02a82c63fb546bb5f23ab2c58e0/src/jsii-project.ts#L299-L300):
   ```ts
   env: {
      NPM_TOKEN: '${{ secrets.NPM_TOKEN }}',
      NPM_DIST_TAG: this.npmDistTag,
   },
   ```
- for [Python in lines 400-401](https://github.com/eladb/projen/blob/ef47eebc3115a02a82c63fb546bb5f23ab2c58e0/src/jsii-project.ts#L400-L401):
   ```ts
   env: {
      TWINE_USERNAME: '${{ secrets.TWINE_USERNAME }}',
      TWINE_PASSWORD: '${{ secrets.TWINE_PASSWORD }}',
   },
   ```

## Proposed solution

Since `jsii-release` already supports setting [`NPM_REGISTRY`](https://github.com/aws/jsii-release#npm) and [`TWINE_REPOSITORY_URL`](https://github.com/aws/jsii-release#pypi) this probably could be managed by allowing one to set values in `.projenrc.js`:
- `options.nodejs.defaultRegistry = false` to enable `NPM_REGISTRY: '${{ secrets.NPM_REGISTRY }}',`
- `options.python.defaultRegistry = false` to enable `TWINE_REPOSITORY_URL: '${{ secrets.TWINE_REPOSITORY_URL }}',`

## Related AWS CodeArtifact issues

Then there's the issue of AWS CodeArtifact, which does not provide permanent token, but a temporary token which has a maximum duration of 12 hours (`43200` seconds). It would require having some extra step in the Github workflow which might not be that easy to support with projen 🤷‍♂️

**Comment by eladb** [comment]
Sounds good! contributions are welcome

**Comment by eladb** [comment]
Let's just add support for specifying the `npmRegistryUrl` and `twineRegistryUrl`. They are not secrets...

Regarding codeArtifact, given we have AWS credentials in a GitHub secret (which I am assuming are required for any solution), the workflow can do whatever it needs (like obtaining a token from aws).

**Comment by JordanSinko** [comment]
@aripalo - Just an FYI... I merged in code to partially accommodate this request with #136 (NPM registry)

**Comment by eladb** [comment]
Resolved by #140
